src/modules/dtfs-storage-client/dtfs-storage-client.service.test.ts

```diff
@@ -0,0 +1,44 @@
+import { ShareFileClient, StorageSharedKeyCredential } from '@azure/storage-file-share';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { resetAllWhenMocks } from 'jest-when';
+
+import DtfsStorageClientService from './dtfs-storage-client.service';
+jest.mock('@azure/storage-file-share');
+
+describe('DtfsStorageFileService', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const baseUrl = valueGenerator.httpsUrl();
+  const accountName = valueGenerator.word();
+  const accountKey = valueGenerator.string();
+  const dealDocumentsFolderPath = valueGenerator.string();
+  const fileLocationPath = valueGenerator.fileLocationPath();
+  const fileName = valueGenerator.fileName();
+  const urlForShareFileClient = `${baseUrl}/${dealDocumentsFolderPath}/${fileLocationPath}/${fileName}`;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    resetAllWhenMocks();
+  });
+
+  describe('constructor', () => {
+    it(`sets the 'storageSharedKeyCredential' property to a new StorageSharedKeyCredential created with the accountName and accountKey`, () => {
+      const dtfsStorageClientService = new DtfsStorageClientService({ baseUrl, accountName, accountKey, dealDocumentsFolderPath });
```

Comment:
Any reason this isn't pulled out into the `beforeEach` block?

Reply:
There was, in that the way I did it first, the tests about the `StorageSharedKeyCredential` were failing because the calls were being reset, but I just realised I can much more simply solve this just by moving the `DtfsStorageClientService` creation after the mock resetting in the `beforeEach` 👍  I've made that change